chia/wallet/puzzles/dao_proposal.clsp

```diff
@@ -266,14 +216,14 @@
   (if self_destruct_flag
       ; assert self_destruct_time > proposal_timelock_length
       ; this is the code path for if we've not been accepted by the treasury for a long time, and we're "bad" for some reason
-      (assert (> self_destruct_time vote_coin_ids_or_proposal_timelock_length)
+      (if (> self_destruct_time vote_coin_ids_or_proposal_timelock_length)
         (list
-          (list CREATE_COIN (calculate_finished_state SINGLETON_STRUCT DAO_FINISHED_STATE_MOD_HASH) ONE (list TREASURY_ID))
+          (list CREATE_COIN (calculate_finished_state (c SINGLETON_MOD_HASH (c SINGLETON_ID LAUNCHER_PUZZLE_HASH)) DAO_FINISHED_STATE_MOD_HASH) ONE (list TREASURY_ID))
```

Comment:
this would be a good place to use `let` to create singleton_struct so we don't construct it twice. Not a showstopper though.
Probably also applies to some of the announcements in this code block